packages/auto-approve/src/valid-pr-schema.json

```diff
@@ -30,7 +30,8 @@
             "yoshi-automation",
             "release-please[bot]",
             "yoshi-code-bot",
-            "gcf-owl-bot[bot]"
```

Comment:
Not related to the changes in this PR, but I noticed several things in this schema definition, so I'm pointing out here.


- Can you add the version of the json schema spec in `$schema`?
- Can you make the `$id` something more unique (e.g. `@google-automations/auto-approve/v1`)?
- `rules` have `additionalProperties`, but it doesn't work for an array.
- I think you should add `additionalProperties`: false on the `aPR` object
- Then you need to add other properties to `properties` definition
- `author` is defined as `enum`, but I think it's too strict. Can it be just a string and validate the value in the app? I also see the benefit of making it very strict, so not a blocker.
- Question: `if` and `then`. Do you have a link to the documentation on `if` and `then`?
- You're restricting the value of the title with a const. This is problematic because when one of the bot (for example googleapis-publisher) changes it's PR title, we'll need to 1) update the schema definition, and 2) update the config files in multiple repositories? Also after we updated the config schema, all the PRs editing auto-approve.yaml will fail because of this schema change. Which seems pretty bad.
- The above point also applied to changedFiles. What if the bot changes the filename?

Reply:
added

- The name `aPR` seems not intuitive because it represents a condition for auto mergeable PRs. How about `autoApproveCondition` or sth?

Reply:
Marking for further discussion.

Reply:
Per our conversation yesterday, here is an [issue](https://github.com/googleapis/repo-automation-bots/issues/2090).